owl/SOMA-IO.owl

```diff
@@ -25,40 +124,321 @@
+            </owl:Restriction>
+        </owl:annotatedTarget>
+        <rdfs:comment xml:lang="en">The file content.</rdfs:comment>
+    </owl:Axiom>
+    <owl:Axiom>
+        <owl:annotatedSource rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Digital_File"/>
+        <owl:annotatedProperty rdf:resource="http://www.w3.org/2000/01/rdf-schema#subClassOf"/>
+        <owl:annotatedTarget>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#hasNameString"/>
+                <owl:someValuesFrom rdf:resource="http://www.w3.org/2001/XMLSchema#string"/>
+            </owl:Restriction>
+        </owl:annotatedTarget>
+        <rdfs:comment xml:lang="en">The path of the file on a computer system.</rdfs:comment>
+    </owl:Axiom>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Executable -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Executable">
+        <owl:equivalentClass>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Computer_Code"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#isGivenMeaningBy"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#ExecutableFormat"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </owl:equivalentClass>
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Computer_Code"/>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#ExecutableFile -->
 
-    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#ComputationalAgent">
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#ExecutableFile">
         <owl:equivalentClass>
             <owl:Class>
                 <owl:intersectionOf rdf:parseType="Collection">
-                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#SoftwareImplementation"/>
-                    <rdf:Description rdf:about="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Agent"/>
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Digital_File"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#realizes"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Executable"/>
+                    </owl:Restriction>
                 </owl:intersectionOf>
             </owl:Class>
         </owl:equivalentClass>
-        <rdfs:comment>An agent that is implemented in software.</rdfs:comment>
-        <rdfs:label>Computational agent</rdfs:label>
+        <rdfs:label xml:lang="en">Executable File</rdfs:label>
```

Comment:
A comment here would be nice.

I would have thought that an ExecutableFile is the realization of either computer code or an algorithm.

Reply:
Added the comment.
See my point above for 'Executable'